src/borough/borough.service.spec.ts

```diff
@@ -20,8 +24,33 @@ describe("Borough service unit", () => {
     boroughService = moduleRef.get<BoroughService>(BoroughService);
   });
 
-  it("service should return a findBoroughsQueryResponseSchema compliant object", async () => {
-    const boroughs = await boroughService.findMany();
-    expect(() => findBoroughsQueryResponseSchema.parse(boroughs)).not.toThrow();
+  describe("findMany", () => {
+    it("service should return a findBoroughsQueryResponseSchema compliant object", async () => {
+      const boroughs = await boroughService.findMany();
+      expect(() =>
+        findBoroughsQueryResponseSchema.parse(boroughs),
+      ).not.toThrow();
+    });
+  });
+
+  describe("findCommunityDistrictsByBoroughId", () => {
+    it("service should return a community districts compliant object", async () => {
+      const { id } = boroughRepositoryMock.checkBoroughByIdMocks[0];
+      const communityDistricts =
+        await boroughService.findCommunityDistrictsByBoroughId(id);
+
+      expect(() =>
+        findCommunityDistrictsByBoroughIdQueryResponseSchema.parse(
+          communityDistricts,
+        ),
+      ).not.toThrow();
+    });
+
+    it("service should throw a resource error when requesting with a missing id", async () => {
```

Comment:
An empty string is different from an id that is not in the database. Empty strings are invalid request parameters. In practical terms, it would never make it to the service. The controller should throw an error before it gets to the service.

To test for a missing id, we want to look at what zod-mock lists and then pick an id that's not in that generated list. This is why we leverage seeding in our mocks- to ensure the generated data is always the same and never accidentally includes the id we designated as "missing"

Reply:
I misunderstood, thanks for explaining. I was thinking missing id as in not there and not as in missing in the db.